packages/blade/src/components/Checkbox/CheckboxGroup/useCheckboxGroup.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import React from 'react';
+import { useControllableState } from '../../../hooks/useControllable';
+import { useId } from '../../../hooks/useId';
+import { useTheme } from '../../BladeProvider';
+import type { CheckboxGroupProps } from './CheckboxGroup';
+
+type UseCheckboxGroupProps = Pick<
+  CheckboxGroupProps,
+  | 'isDisabled'
+  | 'labelPosition'
+  | 'validationState'
+  | 'neccessityIndicator'
+  | 'name'
+  | 'value'
+  | 'defaultValue'
+  | 'onChange'
+>;
+
+const useCheckboxGroup = ({
+  value,
+  defaultValue,
+  isDisabled,
+  labelPosition,
+  onChange,
+  neccessityIndicator,
+  validationState,
+  name,
+}: UseCheckboxGroupProps) => {
+  const { platform } = useTheme();
+  const idBase = useId('checkbox-group');
+  const labelId = `${idBase}-label`;
+  const [checkedValues, setValue] = useControllableState({
+    value,
+    defaultValue: defaultValue || [],
+    onChange,
+  });
+
+  const state = React.useMemo(() => {
```

Comment:
why do you need memo here? why can't these functions be directly declared?

Reply:
Since passing plain objects will change it's reference each render causing unnecessary rerenders. Thus we memo it to only change based on these props.